libs/coin-modules/coin-cosmos/src/config.ts

```diff
@@ -0,0 +1,185 @@
+    type: "object",
+    default: {
+      lcd: "https://api.nyx.nodes.guru",
+      minGasPrice: 0,
+      status: {
+        type: "active",
+      },
+    },
+  },
+  config_currency_onomy: {
+    type: "object",
+    default: {
+      lcd: "https://rest-mainnet.onomy.io",
+      minGasPrice: 0.003,
+      ledgerValidator: "onomyvaloper1fgklp9hemczlwtqp9jqzq3xahh38hznxu9mtmf",
+      status: {
+        type: "active",
+      },
+    },
+  },
+  config_currency_osmo: {
+    type: "object",
+    default: {
+      lcd: "https://osmosis-api.polkachu.com",
+      minGasPrice: 0.025,
+      ledgerValidator: "osmovaloper17cp6fxccqxrpj4zc00w2c7u6y0umc2jajsyc5t",
+      status: {
+        type: "active",
+      },
+    },
+  },
+  config_currency_persistence: {
+    type: "object",
+    default: {
+      lcd: "https://rest.core.persistence.one",
+      minGasPrice: 0.025,
+      ledgerValidator: "persistencevaloper1fgklp9hemczlwtqp9jqzq3xahh38hznxatty38",
+      status: {
+        type: "active",
+      },
+    },
+  },
+  config_currency_quicksilver: {
+    type: "object",
+    default: {
+      lcd: "https://lcd.quicksilver.zone",
+      minGasPrice: 0.0025,
+      ledgerValidator: "quickvaloper1fgklp9hemczlwtqp9jqzq3xahh38hznx02n4pp",
+      status: {
+        type: "active",
+      },
+    },
+  },
+  config_currency_secret_network: {
+    type: "object",
+    default: {
+      lcd: "https://lcd.secret.express",
+      minGasPrice: 0.25,
+      status: {
+        type: "active",
+      },
+    },
+  },
+  config_currency_sei_network: {
+    type: "object",
+    default: {
+      lcd: "https://sei-api.polkachu.com",
+      minGasPrice: 0.1,
+      status: {
+        type: "active",
+      },
+    },
+  },
+  config_currency_stargaze: {
+    type: "object",
+    default: {
+      lcd: "https://stargaze-api.polkachu.com",
+      minGasPrice: 1,
+      status: {
+        type: "active",
+      },
+    },
+  },
+  config_currency_umee: {
+    type: "object",
+    default: {
+      lcd: "https://umee-api.polkachu.com",
+      minGasPrice: 0.1,
+      status: {
+        type: "active",
+      },
+    },
+  },
+  config_currency_coreum: {
+    type: "object",
+    default: {
+      lcd: "https://full-node.mainnet-1.coreum.dev:1317",
+      minGasPrice: 0.1,
+      status: {
+        type: "active",
+      },
+    },
+  },
+  config_currency_injective: {
+    type: "object",
+    default: {
+      lcd: "https://injective-api.polkachu.com",
+      minGasPrice: 900000000,
+      ledgerValidator: "injvaloper1ntn4j2lsu3k60g8xj9pqshqvdj2q5tygyvczpy",
+      status: {
+        type: "active",
+      },
+    },
+  },
+};
+
+import { CurrencyConfig } from "@ledgerhq/coin-framework/config";
+import { MissingCoinConfig } from "@ledgerhq/coin-framework/errors";
+import { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
+
+export type CosmosCoinConfig = CurrencyConfig & CosmosConfig;
+
+export type CoinConfig = (currency: CryptoCurrency) => CosmosCoinConfig;
```

Comment:
You can remove a lot of this code (`setCoinConfig` and `getCoinConfig` implementation) by using the helper `buildConfig`. Just import:

```ts
import buildConConfig, { type CurrencyConfig } from "@ledgerhq/coin-framework/config";
```

Take a look at `coin-xrp` for an example.

Reply:
Done here, wdyt? https://github.com/LedgerHQ/ledger-live/pull/7636/commits/26f3ef27742ec4f1d920213a0fd412c3e4aa957b

As the getter of the coinConfig takes a currency parameters I had to add it to the coin-framework `CoinConfig` and `getCoinConfig` types as an optional parameter.
https://github.com/LedgerHQ/ledger-live/pull/7636/files#diff-1b51549f539752e952a31488c8f9a6d66d934206b5512b103046e373ba423c98R26